### ISSUE TITLE
Default `ls` colors are not readable on a light background

### ISSUE THREAD

**Issue by wavebeem** [nushell/nushell#5961]
### Describe the bug

When using a terminal with a light background, the default colors for `ls` are not readable.

### How to reproduce

1. Use a terminal with a light background
2. Run `ls`

### Expected behavior

I would expect a program to not assume that "white" text is readable against the terminal background by default.

### Screenshots

![image](https://user-images.githubusercontent.com/176815/177436969-bb9cfd46-838b-4df0-b96c-011cb063b822.png)

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.65.0                                          |
| branch             |                                                 |
| commit_hash        | 390d06d4e7383d9fee814713bda8d1aaca858c5a        |
| build_os           | windows-x86_64                                  |
| build_target       | x86_64-pc-windows-msvc                          |
| rust_version       | rustc 1.62.0 (a8314ef7d 2022-06-27)             |
| rust_channel       | stable-x86_64-pc-windows-msvc                   |
| cargo_version      | cargo 1.62.0 (a748cf5a3 2022-06-08)             |
| pkg_version        | 0.65.0                                          |
| build_time         | 2022-07-05 17:33:16 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

### Additional context

Users with astigmatism like myself may prefer light themes for accessibility reasons, and when I looked up [themes info](https://www.nushell.sh/book/coloring_and_theming.html#attributes) for nu it didn't seem very helpful.

**Comment by WindSoilder** [comment]
Hi, you can try the following to make nu friendly to light background:
1. in `$nu.config-path` file, change the relative color config value to `$light_theme` (https://github.com/nushell/nushell/blob/main/docs/sample_config/default_config.nu#L240)
2. Using [vivid](https://www.nushell.sh/book/coloring_and_theming.html#ls-colors-colors-for-the-ls-command) to change LS_COLORS,

**Comment by wavebeem** [comment]
Thanks, changing `$dark_theme` to `$light_theme` in the config file made everything readable. I don't really need to edit LS_COLORS now.

My suggestion is light vs dark theme would be a nice question to ask on the first run wizard for `nu`, or the default colors could be more theme agnostic.

If not, perhaps a short example of switching from dark theme to light theme on the [docs page](https://www.nushell.sh/book/coloring_and_theming.html) would be nice.

I thought I was going to have to make a whole theme from scratch to use nu.

**Comment by WindSoilder** [comment]
@wavebeem Thanks for your suggestion, the light theme relative doc is updated here: https://www.nushell.sh/book/coloring_and_theming.html#working-on-light-background-terminal

**Comment by DavidSouther** [comment]
The docs help, but now colors are illegible are dark mode when the terminal is set to track OS theme. Whether this or a new issue, color_config should be able to track the terminal or os theme preferences.

**Comment by wavebeem** [comment]
@DavidSouther I don't think there's any reliable way to tell what the background color of a given terminal is, which is why I suggest not using "black" or "white" text without also setting a background color. In my experience "bold" text is usually a decent alternative, or just letting the terminal use its default color.

**Comment by DavidSouther** [comment]
I mean, it's 2022, why are we going through LS_COLORS contortions in the first place!? When I open warp, I want to say "use system theme", and I want to get sane colors from the terminal while it's at it without needing to go through contortions. The literal background color of the terminal that the shell is drawing on is secondary (to me, as a user). Maybe there's a hack with a $pre-prompt script that does one of

```
defaults read -g AppleInterfaceStyle # nil or @dark

Get-ItemProperty -Path HKCU:\SOFTWARE\Microsoft\Windows\CurrentVersion\Themes\Personalize -Name AppsUseLightTheme # 0 or 1

gsettings get org.freedesktop.appearance color-scheme # default prefer-dark prefer-light
```

**Comment by DavidSouther** [comment]
Yeah, I got this working:

```
# config.nu

def read_system_theme [] {
  # Replace with Get-ItemProperty or gsettings, as necessary
  do -i { defaults read -globalDomain AppleInterfaceStyle } | str trim
}

def theme [] {
  let theme = (read_system_theme);
  if ($theme == "Dark") {  $dark_theme } else {  $light_theme };
}

config = {
# ...
  hooks: {
    pre_prompt: [{
      let-env config = ($env.config | upsert color_config (theme) )
    }]
}

**Comment by lheckemann** [comment]
Wouldn't it be better for the default to not set a colour at all for anything that's not supposed to be "highlighted" in a particular colour? That works regardless of theming decisions and without extra config on the part of the users (which is important if nushell is to be used for scripts that users who don't otherwise use nushell might want to run).

**Comment by mbhutton** [comment]
For some themes, the outcome is more extreme, e.g. as a new user trying out one of the examples locally, it got the false impression that nutshell didn't work on my machine, as all the important data was invisible:
<img width="594" alt="image" src="https://github.com/nushell/nushell/assets/63345/1b88a7d9-305f-490b-bdea-2de8b11ac65f">

@lheckemann 's suggestion of using the default colour there sounds ideal, as it would work automatically on light and dark themes.

If not changing the default, then supporting a colour value of `terminal_default` or `bold` as an opt-in would also work well.

As a related suggestion, the option to specify the value `reverse` works well for configuring background colours in a way that remains legible for any theme (see lazygit's and lazydocker's `selectedLineBgColor=reverse`).

**Comment by cptpiepmatz** [comment]
@Bahex I think we could select the proper theme using `term query`, no?

**Comment by Bahex** [comment]
@cptpiepmatz Yeah there is a script that @fdncred and I made that queries all the colors, I should probably put it on nu_scripts.

Here it is:
```nushell
# helper function to parse 16bit color
def parse_16bit_color [color: string] {
    let rgb = $color
    | split row /
    | each { into int -r 0x10 | bits and 0x00ff | into binary -c }
    | first 3

    {
        red: ($rgb.0 | encode hex --lower)
        green: ($rgb.1 | encode hex --lower)
        blue: ($rgb.2 | encode hex --lower)
        hex: $"#($rgb | bytes collect | encode hex --lower)"
        rgb: $"RGB(char lp)($rgb | into int | str join ', ')(char rp)"
    }
}

# get all colors in the terminal
export def get-terminal-colors [] {
    let responses = (
        (10..=11 | each { $"($in);?" }) # fg & bg
        | append (0..=15 | each { $"4;($in);?" }) # colors
        | each { term query $"(ansi osc)($in)(ansi st)" --prefix $'(ansi osc)($in | str substring 0..-2)' --terminator (ansi st) | decode }
        | parse $'rgb:{color}'
        | each { parse_16bit_color $in.color }
    )
    {
        fg: ($responses.0)
        bg: ($responses.1)
        colors: ($responses | skip 2)
    }
}
```

There is also a [script on discord](https://discord.com/channels/601130461678272522/615253963645911060/1346668951974707291) that sets the theme automatically based on the background color that calculates perceived lightness.

**Comment by fdncred** [comment]
But not every terminal supports this type of query. So, it would work for some but not others.

**Comment by cptpiepmatz** [comment]
We could try to check if a terminal has a light background and otherwise use our current theme. I think that will be the best we can ever reach

**Comment by fdncred** [comment]
> We could try to check if a terminal has a light background ...

I'm not sure we can. For instance, if you use MacOS and use Terminal.app, the script hangs and you have to ctrl-q the window to force exit it.

**Comment by lheckemann** [comment]
What's wrong with not setting a colour at all?

**Comment by lheckemann** [comment]
I mean to not set a colour on text that has the "primary" colour, not to disable all colour.

**Comment by fdncred** [comment]
We do support \e[39 which we call `ansi default` however, I'm not sure where it would be used because all nushell values should have colors and the repl text should have colors.
